packages/postgres/src/pool.ts

```diff
@@ -870,10 +884,15 @@ export class PostgresPool {
               break;
             }
 
+            if (rowKeys === null) {
+              rowKeys = Object.keys(rows[0] ?? {});
+            }
+            const flattened =
```

Comment:
I couldn't find an obvious way to get the shape of the result, so this differs from doing a `results.fields.length === 1`.